system-tests/tests/load/cre/workflow_don_load_test.go

```diff
@@ -386,50 +372,68 @@ func TestLoad_Workflow_Streams_MockCapabilities(t *testing.T) {
 		}
 	} else {
 		for i := range setupOutput.nodeOutput[1].CLNodes {
-			// TODO: This is brittle, switch to checking the node label
-			if i == 0 { // Skip bootstrap node
-				continue
-			}
 			mockClientsAddress = append(mockClientsAddress, fmt.Sprintf("%s-%s-%d-mock.main.stage.cldev.sh:443", in.Infra.CRIB.Namespace, setupOutput.nodeOutput[1].NodeSetName, i-1))
 		}
 	}
 
+	require.NotEmpty(t, mockClientsAddress, "Could not create mock capability client addresses")
+
 	// Use insecure gRPC connection for local Docker containers. For AWS, use TLS credentials
 	// due to ingress requirements, as grpc.insecure.NewCredentials() doesn't work properly with AWS ingress
 	useInsecure := false
 	if in.Infra.InfraType == "docker" {
 		useInsecure = true
 	}
 
-	require.NoError(t, mocksClient.ConnectAll(mockClientsAddress, useInsecure, true), "could not connect to mock capabilities")
+	require.NoError(t, mocksClient.ConnectAll(mockClientsAddress, useInsecure, cacheClients), "could not connect to mock capabilities")
 
 	testLogger.Info().Msg("Hooking into mock executable capabilities")
 
 	receiveChannel := make(chan capabilities.CapabilityRequest, 1000)
 	require.NoError(t, mocksClient.HookExecutables(ctx, receiveChannel), "could not hook into mock executable")
 
+	// Wait for the remote capability to be exposed, we check if the streams-trigger has subscribers
+	require.NoError(t, mocksClient.WaitForTriggerSubscribers(ctx, "streams-trigger@2.0.0", time.Minute*5), "error while waiting for trigger subscribers")
+
 	labels := map[string]string{
-		"go_test_name": "test1",
+		"go_test_name": "workflow-don-load-test",
 		"branch":       "profile-check",
 		"commit":       "profile-check",
 	}
 
-	g, err := wasp.NewProfile().
-		Add(wasp.NewGenerator(&wasp.Config{
-			CallTimeout: time.Minute * 5, // Give enough time for the workflow to execute
-			LoadType:    wasp.RPS,
-			Schedule: wasp.Combine(
-				wasp.Plain(4, 120*time.Minute),
-			),
-			Gun:                   NewStreamsGun(mocksClient, kb, feedsAddresses, "streams-trigger@2.0.0", receiveChannel, 500, 1),
-			Labels:                labels,
-			LokiConfig:            wasp.NewEnvLokiConfig(),
-			RateLimitUnitDuration: time.Minute,
-		})).
-		Run(false)
-	require.NoError(t, err, "wasp load test did not finish successfully")
+	generator, err := wasp.NewGenerator(&wasp.Config{
+		T:           t,
+		CallTimeout: time.Minute * 2, // Give enough time for the workflow to execute
+		LoadType:    wasp.RPS,
+		Schedule: wasp.Combine(
+			wasp.Plain(4, 5*time.Minute),
+		),
+		Gun:    NewStreamsGun(mocksClient, kb, feedsAddresses, "streams-trigger@2.0.0", receiveChannel, 500, 1),
+		Labels: labels,
+		// LokiConfig:            wasp.NewEnvLokiConfig(),
```

Comment:
remove or add comment when it should be uncommented?

Reply:
Added comment, we will set it up once we have observability working